Hide username in logs #241

jmenezesdev-git opened the pull request:
Initial commit with tests. Replaces "<DriveLetter>:\Users\<Username>\" with "%userprofile%\". Also applies to '/' seperators.
Aims to solve issue #241

Not sure about adding regex to this project so avoided it. 
Exposed a new function in log.h for testing.
Testing & .h changes could be skipped.

DankRank commented:
some possible improvements and things to note
- in is_drive_letter, use character constants `(c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z')`
- `/` and `\` can be mixed freely in the path, they aren't necessarily all `/` or all `\`
- on Wine (linux), home directory is `Z:\home\<username>\`
- on XP it's `C:\Documents and Settings\<username>\` I think

jmenezesdev-git commented:
@DankRank 
After looking through the Linux documentation for this project it seems like users are asked to just use Wine which enforces Windows conventions for folder names.
I just pushed a commit that addresses all your suggestions.

@brliron 
You can also launch and run Touhou games from user directories that are not your own. IE C:\Users\Public\TH\... vs C:\Users\<username>\TH. I'll hold off on implementing GetUserProfileDirectoryW until zero and yourself make a decision.

brliron commented:
> Very unlikely, but it could be a directory of another real user. Not sure if such case is worth considering though.
It isn't worth considering IMO.

> Unless the user explicitly installs and runs stuff in `.wine/drive_c`
It might happen if they run the official installer with Wine... but it would end up in `C:\Program Files\...` in the thcrap logs so we don't care.

> Maybe we should make a web tool that runs a simple regexp on the file contents, to side step the issue entirely.
The problem with this approach is making that web tool known.
... I guess a mention at the top of the log file could be enough? It would also help to bring back a Discord bot with custom commands, so that we can type `!logs` in #support and have a small message with where the logs are and the url of that tool.
